### PR TITLE
First pass at file api for k8s

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Adds the driver interface to https://github.com/FlowFuse/nr-launcher/issues/273

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
@Steve-Mcl This and the docker one, please review but don't merge

**Review by Steve-Mcl — COMMENTED** [review]
couple of point noted

File: kubernetes.js [hunk 1]
```diff
@@ -498,6 +499,11 @@ const getEndpoints = async (project) => {
     }
 }
 
+const getStaticFileUrl = async (project, filePath) => {
+    const prefix = project.safeName.match(/^[0-9]/) ? 'srv-' : ''
+    return `http://${prefix}${project.safeName}.${this._namespace}:2880/flowforge/files/_/${filePath}`
```
[the remark this answers is not in the document]
Comment (hardillb) [reply kubernetes.js:504]: The whole file uses projects, I'm not mixing them and I'm not going to use this PR to change the rest

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "dependencies": {
         "@aws-sdk/client-efs": "^3.600.0",
         "@kubernetes/client-node": "^0.18.1",
+        "form-data": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (hardillb) [reply package.json:26]: OK, we need to review this across all the static asset code, as Nick has used it in the localfs driver

[the remark this answers is not in the document]
Comment (hardillb) [reply package.json:26]: @Steve-Mcl I'm going to leave this for now as it works

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply package.json:26]: ok, understand. Should probably raise an issue to mop this up.

**Review by Steve-Mcl — APPROVED** [review]

**Pull request merged**